refactor: move IDisposable to core interfaces (#16184)
microsoft/FluidFramework: common/lib/common-definitions/CHANGELOG.md

# @fluidframework/common-definitions Changelog

## [1.2.0](https://github.com/microsoft/FluidFramework/releases/tag/common-definitions_v1.2.0)

### Deprecated interface

> The following interface is deprecated in this release. The implementations have been moved to other
> packages.

Reviewer: Nit. Just to make it clear it's the interfaces themselves that are moving, not classes that implement the interfaces.
```suggestion
The following interface is deprecated in this release. Its replacement has been moved to another package.
```
